pkg/webhook/pod/tikv_creater.go

```diff
@@ -84,3 +89,37 @@ func (pc *PodAdmissionControl) admitCreateTiKVPod(pod *core.Pod, tc *v1alpha1.Ti
 
 	return util.ARSuccess()
 }
+
+// This is method is to make compatible between old pdapi version and 4.0 pdapi version.
+// To get more detail, see: https://github.com/pingcap/tidb-operator/pull/1831
+func filterLeaderEvictScheduler(evictLeaderSchedulers []string, pdClient pdapi.PDClient) (sets.String, error) {
+	schedulerIds := sets.String{}
+	if len(evictLeaderSchedulers) == 1 && evictLeaderSchedulers[0] == evictSchedulerLeader {
+		c, err := pdClient.GetConfig()
+		if err != nil {
+			return schedulerIds, err
+		}
+		if c.Schedule != nil {
+			if c.Schedule.SchedulersPayload == nil {
```

Comment:
If it's nil, why still need to continue to the following codes?

Reply:
Not necessary,  just solve the nil or empty payload with the same method.